### ISSUE TITLE
AssertionError when using hypothesis with threading

### ISSUE THREAD

**Issue by Ezibenroc** [HypothesisWorks/hypothesis-python#337]
I would like to use hypothesis with `threading` module.
The following code raises an `AssertionError`:

``` python
import hypothesis
import hypothesis.strategies as st
import threading

@hypothesis.given(st.integers())
def test(n):
    pass

t = threading.Thread(target = test)
t.start()
t.join()
```

Am I missing something? How can I run a function which uses hypothesis in a thread?

**Comment by pajowu** [comment]
Looks like, [_settings.py line 94](https://github.com/HypothesisWorks/hypothesis-python/blob/master/src/hypothesis/_settings.py#L94) should't reassign `default_variable.value`.
`settings.load_profile` never returns anything, so code could be simplyfied down to
```
def load_profile():
    pass
value = load_profile()
assert value is not None
```

**Comment by DRMacIver** [comment]
Huh. Sorry about that. Victim of the fact that due to reasons the settings module is excluded from my 100% coverage requirements I guess. :-(

I'm travelling at the moment and what free time I have is quite occupied, so I'll probably not get a chance to look at this soon. I'd happily accept and review a pull request (it's easier to do that from a phone!) but if not I'll try to get to fixing this in the next week or two.

**Comment by Zac-HD** [comment]
Closed by 1bcbc4eea9c64816d22465a11e5cf1b919c5d8fe?

**Comment by Ezibenroc** [comment]
@Zac-HD Wow, this commit is quite old, I did not notice it.
I tested again. The problem seems to be fixed indeed.

I am closing the issue, thanks for your help.
